app/Models/Achievement.php

```diff
@@ -351,6 +353,28 @@ public function playerAchievements(): HasMany
         return $this->hasMany(PlayerAchievement::class, 'achievement_id', 'ID');
     }
 
+    /**
+     * @return HasMany<EventAchievement>
+     */
+    public function eventAchievements(): HasMany
+    {
+        return $this->hasMany(EventAchievement::class, 'source_achievement_id', 'ID');
+    }
+
+    public function eventData(): EventAchievement
+    {
+        return $this->hasOne(EventAchievement::class, 'achievement_id')->first();
+    }
+
+    /**
+     * An achievement should have at most one "event data", but this is needed for the management panel
+     * @return HasMany<EventAchievement>
+     */
+    public function eventDatas(): HasMany
+    {
+        return $this->hasMany(EventAchievement::class, 'achievement_id', 'ID');
+    }
```

Comment:
`eventData()` is a query, not a relationship. Judging by the comment for `eventDatas()`, the definition of `eventData()` should probably look more like this:
```php
/**
 * @return HasOne<EventAchievement>
 */
public function eventData(): HasOne
{
    return $this->hasOne(EventAchievement::class, 'achievement_id');
}
```

`eventDatas()` can likely be removed altogether, due to a comment I will make elsewhere.